src/core/input_descriptor.rs

```diff
@@ -662,67 +660,57 @@ impl ConstraintsField {
-            // NOTE: It may not be a given that the last path is the field name.
-            // TODO: Cannot use the field path as a unique property, it may be associated to different
-            // credential types.
-            // NOTE: Include the namespace for uniqueness of the requested field type.
-            .filter_map(|path| path.split(&['-', '.', ':', '@'][..]).last())
-            .map(|path| {
-                path.chars()
-                    .fold(String::new(), |mut acc, c| {
-                        // Convert camelCase to space-separated words with capitalized first letter.
-                        if c.is_uppercase() {
-                            acc.push(' ');
-                        }
-
-                        // Check if the field is snake_case and convert to
-                        // space-separated words with capitalized first letter.
-                        if c == '_' {
-                            acc.push(' ');
-                            return acc;
-                        }
-
-                        acc.push(c);
-                        acc
-                    })
-                    // Split the path based on empty spaces and uppercase the first letter of each word.
-                    .split(' ')
-                    .fold(String::new(), |desc, word| {
-                        let word =
-                            word.chars()
-                                .enumerate()
-                                .fold(String::new(), |mut acc, (i, c)| {
-                                    // Capitalize the first letter of the word.
-                                    if i == 0 {
-                                        if let Some(c) = c.to_uppercase().next() {
-                                            acc.push(c);
-                                            return acc;
-                                        }
-                                    }
-                                    acc.push(c);
-                                    acc
-                                });
-
-                        format!("{desc} {}", word.trim_end())
-                    })
-                    .trim_end()
-                    .to_string()
-            })
-            .collect()
+            .filter_map(|path| selector(path).ok())
+            .flatten()
+            .map(ToOwned::to_owned)
+            // NOTE: It is likely that only one of the paths will
+            // match the selector. Therefore, we're selecting only
+            // the first match that exists, if any.
+            //
+            // It may also be possible that multiple paths match,
+            // in which case there may be multiple raw fields
+            // that can be requested.
+            //
+            // As a result, it may be more acceptable to use `collect()` instead
+            // of `first()` to return all the possible requested fields.
+            .collect::<Vec<serde_json::Value>>()
+            .pop();
```

Comment:
If it's possible that more than one paths match then we should return all of those values. The library needs to be as flexible and correct as possible. We can decide what to do with the result in our implementations downstream.
```suggestion
            .collect::<Vec<serde_json::Value>>();
```